### PR TITLE
Saved searches

### LINKED ISSUE #49: Support saving multiple search sets to user profile
User profiles have a set of named search parameter sets
Users can CRUD these parameters

### PR DESCRIPTION
**Pull request by afrasier**
This PR addresses #49 and #54 (Don't put "Resolves" in front of those numbers unless you want the PR to automatically close them :))

In addition to new functionality, this PR refactors most all of the current URLs into a more REST-ful, resource based approach. A table of the refactoring is below: ( @mjoyce91 )

| Endpoint (current)               	| Endpoint (new)                  	|
|----------------------------------	|---------------------------------	|
| /api/v1/accounts/profile/        	| /api/v1/profile/                	|
| /api/v1/language/proficiencies/  	| /api/v1/language_proficiency/   	|
| /api/v1/language/qualifications/ 	| /api/v1/language_qualification/ 	|
| /api/v1/organization/posts/      	| /api/v1/post/                   	|
| /api/v1/organization/tod/        	| /api/v1/tour_of_duty/           	|
| /api/v1/position/grades/         	| /api/v1/grade/                  	|
| /api/v1/position/skills/         	| /api/v1/skill/                  	|
| /api/v1/accounts/share/          	| /api/v1/share/                  	|

All other endpoints remain the same.

Changes:
* Created `SavedSearch` object, which stores a set of filters, endpoint, and name
* Removed `preferred_position` data from `UserProfile` as saved searches supersedes it
* Added the following endpoints to support storing, retrieving, and deleting saved searches

| Methods            | Endpoint               |
|--------------------|------------------------|
| PATCH, DELETE, GET | /api/v1/searches/{id}/ |
| POST, GET          | /api/v1/searches/      |

* Added test to test new saved search functionality
* Added validation to saved search serializers to ensure stored filters are valid for the filter set and endpoint
* Refactored all endpoints to be more RESTful
* Updated tests to reflect new URL structure

### CODE REVIEW

**Comment by burgwyn** [comment]
Do we want to refer to these resources in singular or plural form?  I do see `/api/v1/accounts/token/` - do we want to change that?

**Comment by afrasier** [comment]
I prefer singular, I left "accounts" alone because that's where all the login mess happens so I didn't want to tinker with it while it's settling on the front end.

File: talentmap_api/position/views.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ class PositionListView(ReadOnlyModelViewSet):
     filter_class = PositionFilter
 
     def get_queryset(self):
+        print(self.request.query_params)
```
Comment (jseppi) [inline talentmap_api/position/views.py]: looks like a forgotten `print` debug line?

File: talentmap_api/user_profile/models.py [hunk 2]
```diff
@@ -47,6 +36,23 @@ class Sharable(models.Model):
...
+    Represents a saved search.
+
+    Filters field contains a JSON object of valid filters and their values, which
+    can be used to reconstruct the get request
+    '''
+    owner = models.ForeignKey(UserProfile, related_name="saved_searches")
+
+    name = models.TextField(default="Saved Search", help_text="The name of the saved search")
+    endpoint = models.TextField(help_text="The endpoint for this search and filter")
+    filters = JSONField(default=dict, help_text="JSON object containing filters representing the saved search")
```
Comment (jseppi) [inline talentmap_api/user_profile/models.py]: The example JSON object comment from the previous `position_preferences` field was helpful, maybe add it back in.

**Pull request merged**